server/app.js

```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
 app.get('/', function(req,res){
     res.redirect('index.html');
 });
- 
-var client = new net.Socket();
+
+require('events').EventEmitter.prototype._maxListeners = 100; //max listeners
+
+var PythonShell = require('python-shell');
+
+var client = new net.Socket();  
+
+client.on('err', function(err){
+            // handle the error safely
+            console.log("Client error");
+            console.log(err)
+        });
 
 io.on('connection',function(socket){ 
+
+    //////////////////////////////////////////////////////////////////
+    // Logging server stuff //
+    var server = net.createServer(function(sock) {
+        sock.on('err', function(err){
+            console.log("vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv\n\n");
+            console.log(err);
+            console.log("\n\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+            server.close();
+            server.listen(5002, '192.168.43.12');  
+        });
+        sock.on('data', function(data) {
+            //console.log(data);
```

Comment:
Instead of commenting out the logging information, would it be better to have a function:
```
function debug_log(string) {
    if (debug_on) {
        console.log(string);
     }
}
```
And have the flag `debug_on` configured at the top of the file.
This way we can turn off logging extra debug messages or not all at once.
If javascript has a simple logging module that can do this that works too.